lightning-background-processor/src/lib.rs

```diff
@@ -107,6 +107,25 @@ ChannelManagerPersister<Signer, M, T, K, F, L> for Fun where
 	}
 }
 
+/// Trait which handles persisting a [`NetworkGraph`] to disk.
+///
+/// [`NetworkGraph`]: lightning::routing::network_graph::NetworkGraph
+pub trait NetworkGraphPersister {
+	/// Persist the given [`NetworkGraph`] to disk, returning an error if persistence failed
+	/// which will cause the [`BackgroundProcessor`] which called this method to exit
+	/// 
+	/// [`NetworkGraph`]: lightning::routing::network_graph::NetworkGraph
+	fn persist_graph(&self, network_graph: &NetworkGraph) -> Result<(), std::io::Error>;
```

Comment:
I don't feel super strongly about it either way, but wanted to at least throw out the idea of just shoving this in the ChannelManagerPersister (and renaming it) instead of having two traits with single functions. I could see some reasons users may want them separate but it does generate a good bit more boilerplate for them and I'm not sure how common that would be.